src/android/io/cozy/plugins/listlibraryitems/ListLibraryItems.java

```diff
@@ -305,46 +298,36 @@ protected Integer doInBackground(UploadFilePayload... uploadFilePayloads) {
                     String val = headers.getString(key);
                     huc.setRequestProperty(key, val);
                 }
-                // urlConnection.setRequestProperty("Connection", "Keep-Alive");
                 huc.setRequestProperty("Content-Length", "" + FILE_SZ);
                 huc.setRequestProperty("User-Agent", System.getProperty("http.agent"));
 
-
                 // config request
                 huc.setUseCaches(false);
                 huc.setDoInput(true);
                 huc.setDoOutput(true);
                 huc.setRequestMethod("POST");
-                // urlConnection.setInstanceFollowRedirects(false);
                 huc.setChunkedStreamingMode(1024 * 1000);
                 huc.setInstanceFollowRedirects(true);
-                // urlConnection.setFixedLengthStreamingMode(bytesAvailable);
-
 
                 // read input file chunks + publish progress
                 OutputStream out = huc.getOutputStream();
-                buffer_size = Math.min(bytes_available, MAX_BUFFER_SZ);
+                buffer_size = 4 * 1024;
                 buffer = new byte[buffer_size];
-                bytes_read = fis.read(buffer, 0, buffer_size);
-                total_bytes = bytes_read;
-                while (bytes_read > 0) {
-                    out.write(buffer, 0, buffer_size);
-                    bytes_available = fis.available();
-                    buffer_size = Math.min(bytes_available, MAX_BUFFER_SZ);
-                    bytes_read = fis.read(buffer, 0, buffer_size);
+                total_bytes = 0;
+                while ((bytes_read = fis.read(buffer, 0, bytes_read)) != -1) {
```

Comment:
`fis.read(buffer, 0, bytes_read)` is buggy, at least must be `fis.read(buffer, 0, buffer_size)`, but better to use directly `fis.read(buffer)` and let java doing the computation internaly